### ISSUE TITLE
cannot use it with current dietpi!

### ISSUE THREAD

**Issue by DrCWO** [meldron/node-alsa-capture#9]
Hello,
I get errors using DietPi v8.10.2.
node was installed by the software selection of dietpi.
Help would be much appreciated :-)

Thanks
```
pi@dietPi:~/rr$ uname -a
Linux dietPi 5.15.61-v8+ #1579 SMP PREEMPT Fri Aug 26 11:16:44 BST 2022 **aarch64** GNU/Linux

pi@dietPi:~/rr$ node -v
v19.0.0

pi@dietPi:~/rr$ npm -v
8.19.2

pi@dietPi:~/rr$ apt-cache policy libasound2
libasound2:
  Installed: 1.2.4-1.1+rpt2
  Candidate: 1.2.4-1.1+rpt2
  Version table:
 *** 1.2.4-1.1+rpt2 500
        500 https://archive.raspberrypi.org/debian bullseye/main arm64 Packages
        100 /var/lib/dpkg/status
     1.2.4-1.1 500
        500 https://deb.debian.org/debian bullseye/main arm64 Packages

pi@dietPi:~/rr$ apt-cache policy libasound2-dev
libasound2-dev:
  Installed: 1.2.4-1.1+rpt2
  Candidate: 1.2.4-1.1+rpt2
  Version table:
 *** 1.2.4-1.1+rpt2 500
        500 https://archive.raspberrypi.org/debian bullseye/main arm64 Packages
        100 /var/lib/dpkg/status
     1.2.4-1.1 500
        500 https://deb.debian.org/debian bullseye/main arm64 Packages

pi@dietPi:~/rr$ npm install alsa-capture --save
npm ERR! code 1
npm ERR! path /home/pi/rr/node_modules/alsa-capture
npm ERR! command failed
npm ERR! command sh -c -- node-gyp rebuild
npm ERR! make: Entering directory '/home/pi/rr/node_modules/alsa-capture/build'
npm ERR!   CXX(target) Release/obj.target/capture/capture.o
npm ERR! make: Leaving directory '/home/pi/rr/node_modules/alsa-capture/build'
npm ERR! gyp info it worked if it ends with ok
npm ERR! gyp info using node-gyp@8.4.1
npm ERR! gyp info using node@19.0.0 | linux | arm64
npm ERR! gyp info find Python using Python version 3.9.2 found at "/usr/bin/python3"
npm ERR! gyp info spawn /usr/bin/python3
npm ERR! gyp info spawn args [
npm ERR! gyp info spawn args   '/home/pi/rr/node_modules/node-gyp/gyp/gyp_main.py',
npm ERR! gyp info spawn args   'binding.gyp',
npm ERR! gyp info spawn args   '-f',
npm ERR! gyp info spawn args   'make',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/pi/rr/node_modules/alsa-capture/build/config.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/pi/rr/node_modules/node-gyp/addon.gypi',
npm ERR! gyp info spawn args   '-I',
npm ERR! gyp info spawn args   '/home/pi/.cache/node-gyp/19.0.0/include/node/common.gypi',
npm ERR! gyp info spawn args   '-Dlibrary=shared_library',
npm ERR! gyp info spawn args   '-Dvisibility=default',
npm ERR! gyp info spawn args   '-Dnode_root_dir=/home/pi/.cache/node-gyp/19.0.0',
npm ERR! gyp info spawn args   '-Dnode_gyp_dir=/home/pi/rr/node_modules/node-gyp',
npm ERR! gyp info spawn args   '-Dnode_lib_file=/home/pi/.cache/node-gyp/19.0.0/<(target_arch)/node.lib',
npm ERR! gyp info spawn args   '-Dmodule_root_dir=/home/pi/rr/node_modules/alsa-capture',
npm ERR! gyp info spawn args   '-Dnode_engine=v8',
npm ERR! gyp info spawn args   '--depth=.',
npm ERR! gyp info spawn args   '--no-parallel',
npm ERR! gyp info spawn args   '--generator-output',
npm ERR! gyp info spawn args   'build',
npm ERR! gyp info spawn args   '-Goutput_dir=.'
npm ERR! gyp info spawn args ]
npm ERR! gyp info spawn make
npm ERR! gyp info spawn args [ 'BUILDTYPE=Release', '-C', 'build' ]
npm ERR! In file included from /home/pi/.cache/node-gyp/19.0.0/include/node/v8-object.h:9,
npm ERR!                  from /home/pi/.cache/node-gyp/19.0.0/include/node/v8-array-buffer.h:13,
npm ERR!                  from /home/pi/.cache/node-gyp/19.0.0/include/node/v8.h:24,
npm ERR!                  from /home/pi/.cache/node-gyp/19.0.0/include/node/node.h:73,
npm ERR!                  from ../../nan/nan.h:60,
npm ERR!                  from ../capture.cc:13:
npm ERR! /home/pi/.cache/node-gyp/19.0.0/include/node/v8-maybe.h:106:45: error: ‘is_lvalue_reference_v’ is not a member of ‘std’; did you mean ‘is_lvalue_reference’?
npm ERR!   106 |   template <class U, std::enable_if_t<!std::is_lvalue_reference_v<U>>*>
npm ERR!       |                                             ^~~~~~~~~~~~~~~~~~~~~
npm ERR!       |                                             is_lvalue_reference
npm ERR! /home/pi/.cache/node-gyp/19.0.0/include/node/v8-maybe.h:106:66: warning: logical not is only applied to the left hand side of comparison [-Wlogical-not-parentheses]
npm ERR!   106 |   template <class U, std::enable_if_t<!std::is_lvalue_reference_v<U>>*>
npm ERR!       |                                                                  ^
npm ERR! /home/pi/.cache/node-gyp/19.0.0/include/node/v8-maybe.h:106:67: error: template argument 1 is invalid
npm ERR!   106 |   template <class U, std::enable_if_t<!std::is_lvalue_reference_v<U>>*>
npm ERR!       |                                                                   ^
npm ERR! /home/pi/.cache/node-gyp/19.0.0/include/node/v8-maybe.h:106:71: error: expected unqualified-id before ‘>’ token
npm ERR!   106 |   template <class U, std::enable_if_t<!std::is_lvalue_reference_v<U>>*>
npm ERR!       |                                                                       ^
npm ERR! /home/pi/.cache/node-gyp/19.0.0/include/node/v8-maybe.h:123:43: error: ‘is_lvalue_reference_v’ is not a member of ‘std’; did you mean ‘is_lvalue_reference’?
npm ERR!   123 | template <class T, std::enable_if_t<!std::is_lvalue_reference_v<T>>* = nullptr>
npm ERR!       |                                           ^~~~~~~~~~~~~~~~~~~~~
npm ERR!       |                                           is_lvalue_reference
npm ERR! /home/pi/.cache/node-gyp/19.0.0/include/node/v8-maybe.h:123:64: warning: logical not is only applied to the left hand side of comparison [-Wlogical-not-parentheses]
npm ERR!   123 | template <class T, std::enable_if_t<!std::is_lvalue_reference_v<T>>* = nullptr>
npm ERR!       |                                                                ^
npm ERR! /home/pi/.cache/node-gyp/19.0.0/include/node/v8-maybe.h:123:65: error: template argument 1 is invalid
npm ERR!   123 | template <class T, std::enable_if_t<!std::is_lvalue_reference_v<T>>* = nullptr>
npm ERR!       |                                                                 ^
npm ERR! /home/pi/.cache/node-gyp/19.0.0/include/node/v8-maybe.h:123:70: error: expected unqualified-id before ‘=’ token
npm ERR!   123 | template <class T, std::enable_if_t<!std::is_lvalue_reference_v<T>>* = nullptr>
npm ERR!       |                                                                      ^
npm ERR! make: *** [capture.target.mk:117: Release/obj.target/capture/capture.o] Error 1
npm ERR! gyp ERR! build error
npm ERR! gyp ERR! stack Error: `make` failed with exit code: 2
npm ERR! gyp ERR! stack     at ChildProcess.onExit (/home/pi/rr/node_modules/node-gyp/lib/build.js:194:23)
npm ERR! gyp ERR! stack     at ChildProcess.emit (node:events:513:28)
npm ERR! gyp ERR! stack     at ChildProcess._handle.onexit (node:internal/child_process:293:12)
npm ERR! gyp ERR! System Linux 5.15.61-v8+
npm ERR! gyp ERR! command "/usr/local/bin/node" "/home/pi/rr/node_modules/.bin/node-gyp" "rebuild"
npm ERR! gyp ERR! cwd /home/pi/rr/node_modules/alsa-capture
npm ERR! gyp ERR! node -v v19.0.0
npm ERR! gyp ERR! node-gyp -v v8.4.1
npm ERR! gyp ERR! not ok

npm ERR! A complete log of this run can be found in:
npm ERR!     /home/pi/.npm/_logs/2022-10-25T10_27_04_769Z-debug-0.log
pi@dietPi:~/rr$



```

**Comment by DrCWO** [comment]
For verification I downgraded to:
https://nodejs.org/dist/latest-v16.x/node-v16.18.0-linux-arm64.tar.gz

```
pi@dietPi:~/rr$ node -v
v16.18.0

pi@dietpi:~/rr$ npm -v
8.19.2

pi@dietPi :~/rr$ npm install alsa-capture --save

added 93 packages, and audited 94 packages in 24s

4 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities
```

This works as expected. Seams to be an issue with latest node...

**Comment by meldron** [comment]
HI @DrCWO, thanks for letting me know! I havn't yet used the library with node 18 or 19. I will take a look

**Comment by meldron** [comment]
I just released version `v0.3.0`, which should work on node 19. Please let me know if there are any problems.

**Comment by DrCWO** [comment]
Works, thanks' a lot 👍
